cc-dashboard-backend/index.js

```diff
@@ -0,0 +1,75 @@
+const functions = require('firebase-functions');
+const admin = require('firebase-admin');
+const Router = require('express');
+const Cors = require('cors')({origin: true});
+
+const rt = Router();
+rt.use(Cors);
+admin.initializeApp();
+const db = admin.firestore();
+
+exports.getFlightCodes = functions.https.onRequest(async (req, res) => {
+  Cors(req, res, async () => {
+    res.set('Access-Control-Allow-Origin', "*")
+    res.set('Access-Control-Allow-Methods', 'GET')
+    const collection = db.collection('squadron');
+    const snapshot = collection.get();
+  
+    if ((await snapshot).empty) {
+      res.status(400).send({result: false, message: "data does not exist"});
+    }
+    let output = [];
+    (await snapshot).forEach(async doc => {
+      output.push(doc.id);
+    })
+  
+    res.status(200).send({flights: output});
+  });
+}) 
+
+rt.get('/:id', async(req, res) => {
+  Cors(req, res, async () => {
+    const collectionRef = await db.collection('squadron');
+    const airmenCollection = await collectionRef.doc(req.params.id).collection('airmen').get();
+  
+    let output = [];
+    airmenCollection.forEach(async airman => {
+      output.push({airmanId: airman.id, airmanData: airman.data()});
+    })
```

Comment:
We could probably get away with using this simplified version:
`let output = await airmenCollection.map( airman => ({airmanId: airman.id, airmanData: airman.data()} );`
 to get the same array.